e2e/cypress/integration/plugins/marketplace/marketplace_spec.js

```diff
@@ -7,19 +7,92 @@
+describe('Plugin Marketplace', () => {
+    after(() => {
+        // # Restore default configuration
+        const newSettings = {
+            PluginSettings: {
+                Enable: true,
+                EnableMarketplace: true,
+                MarketplaceUrl: 'https://api.integrations.mattermost.com',
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+    });
+
+    describe('should not render in main menu', () => {
+        afterEach(() => {
+            // # Click hamburger main menu
+            cy.get('#sidebarHeaderDropdownButton').click();
+
+            // * Dropdown menu should be visible
+            cy.get('#sidebarDropdownMenu').should('be.visible');
+
+            // * Marketplace button should not be visible
+            cy.get('#marketplaceModal').should('not.be.visible');
+        });
+
+        it('for non-admin', () => {
+            // # Configure marketplace as enabled
+            const newSettings = {
+                PluginSettings: {
+                    Enable: true,
+                    EnableMarketplace: true,
+                    MarketplaceUrl: 'https://api.integrations.mattermost.com',
+                },
+            };
+            cy.apiUpdateConfig(newSettings);
+
+            // # Login as non admin user
+            cy.apiLogin('user-1');
+            cy.visit('/');
+        });
+
+        it('when marketplace disabled', () => {
+            // # Configure marketplace as disabled
+            const newSettings = {
+                PluginSettings: {
+                    Enable: true,
+                    EnableMarketplace: false,
+                    MarketplaceUrl: 'https://api.integrations.mattermost.com',
+                },
+            };
+            cy.apiUpdateConfig(newSettings);
+
+            // # Login as sysadmin
+            cy.apiLogin('sysadmin');
+            cy.visit('/');
+        });
+
+        it('when plugins disabled', () => {
+            // # Configure plugins as disabled
+            const newSettings = {
+                PluginSettings: {
+                    Enable: false,
+                    EnableMarketplace: true,
+                    MarketplaceUrl: 'https://api.integrations.mattermost.com',
+                },
+            };
+            cy.apiUpdateConfig(newSettings);
+
+            // # Login as sysadmin
+            cy.apiLogin('sysadmin');
+            cy.visit('/');
+        });
+    });
+
+    it('should render an error bar when connecting to an invalid marketplace server', () => {
         // # Set ServiceSettings to expected values
         const newSettings = {
             PluginSettings: {
                 Enable: true,
-                EnableUploads: true,
                 EnableMarketplace: true,
+                MarketplaceUrl: 'example.com',
             },
         };
         cy.apiUpdateConfig(newSettings);
 
-        // # Login as sysadmin
+        // # Login as non admin user
         cy.apiLogin('sysadmin');
```

Comment:
The comment says `non admin user`, though we are logging in as `sysadmin`. Is this intentional?

Reply:
Good catch -- fixing the comment.